sourmash/cli/__init__.py

```diff
@@ -0,0 +1,112 @@
+        args = super(SourmashParser, self).parse_args(args=args, namespace=namespace)
+        if ('quiet' not in args or not args.quiet) and self.citation:
+            self.print_citation()
+
+        # BEGIN: dirty hacks to simultaneously support new and previous interface
+        if hasattr(args, 'subcmd') and args.subcmd == 'import':
+            args.subcmd = 'ingest'
+        if hasattr(args, 'cmd') and args.cmd == 'sbt_combine':
+            args.cmd = 'sbt'
+            args.subcmd = 'combine'
+        if hasattr(args, 'cmd') and args.cmd == 'index':
+            args.cmd = 'sbt'
+            args.subcmd = 'index'
+        if hasattr(args, 'cmd') and args.cmd == 'categorize':
+            args.cmd = 'sbt'
+            args.subcmd = 'categorize'
+        if hasattr(args, 'cmd') and args.cmd == 'watch':
+            args.subcmd = 'sbt'
+            args.subcmd = 'watch'
+        if hasattr(args, 'subcmd') and args.subcmd == 'compare_csv':
+            args.subcmd = 'compare'
+        # END: dirty hacks to simultaneously support new and previous interface
+        return args
+
+
+def add_moltype_args(parser):
+    parser.add_argument(
+        '--protein', dest='protein', action='store_true',
+        help='choose a protein signature; by default, a nucleotide signature is used'
+    )
+    parser.add_argument(
+        '--dayhoff', dest='dayhoff', action='store_true',
+        help='build Dayhoff-encoded amino acid signatures'
+    )
+    parser.add_argument(
+        '--hp', '--hydrophobic-polar', dest='hp', action='store_true',
+        help='build hydrophobic-polar-encoded amino acid signatures'
+    )
+
+
+def add_ksize_arg(parser, default=21):
+    parser.add_argument(
+        '-k', '--ksize', metavar='K', default=None, type=int,
+        help='k-mer size; default={d}'.format(d=default)
+    )
+
+
+def get_parser():
+    commands = ['compute', 'compare', 'search', 'plot', 'gather', 'lca', 'sbt', 'info', 'signature']
+    commandstr = ' -- '.join(sorted(commands))
+
+    desc = 'Compute, compare, manipulate, and analyze MinHash sketches of DNA sequences.'
+    parser = SourmashParser(prog='sourmash', description=desc)
+    parser._optionals.title = 'Options'
+    parser.add_argument('-v', '--version', action='version', version='sourmash '+ VERSION)
+    parser.add_argument('-q', '--quiet', action='store_true', help='don\'t print citation information')
+    sub = parser.add_subparsers(
+        title='Commands', dest='cmd', metavar='cmd', help=commandstr,
+        description='Invoke "sourmash <cmd> --help" for more details on executing each command.'
+    )
+    for cmd in commands:
+        getattr(sys.modules[__name__], cmd).subparser(sub)
+    # BEGIN: dirty hacks to simultaneously support new and previous interface
+    sbt.categorize.subparser(sub)
+    sbt.combine.alt_subparser(sub)
+    sbt.index.subparser(sub)
+    sbt.watch.subparser(sub)
+    # END: dirty hacks to simultaneously support new and previous interface
```

Comment:
Moar dirty hacks: support both `sourmash categorize` and `sourmash sbt categorize`, both `sourmash sbt_combine` and `sourmash sbt combine`, and so on.

Reply:
`categorize` is potentially an [index operation](https://github.com/dib-lab/sourmash/pull/556#issuecomment-449001254), should work with `lca` too (so I suggest keeping it as a upper level, instead of inside `sbt`).

Reply:
Gotcha. Well, that only leaves the `sbt_combine` as a candidate for grouping sbt-specific subcommands, so I'll just undo that grouping and revert those 4 subcommands to top-level commands.